x-pack/plugins/endpoint/server/routes/resolver/queries/stats.ts

```diff
@@ -0,0 +1,151 @@
+                    // event part of the filter
+                    {
+                      bool: {
+                        filter: [
+                          { term: { 'event.kind': 'event' } },
+                          { terms: { 'endgame.unique_pid': uniquePIDs } },
+                          {
+                            bool: {
+                              must_not: {
+                                term: { 'event.category': 'process' },
+                              },
+                            },
+                          },
+                        ],
+                      },
+                    },
+                    // alert part of the filter
+                    {
+                      bool: {
+                        filter: [
+                          { term: { 'event.kind': 'alert' } },
+                          {
+                            terms: {
+                              'endgame.data.alert_details.acting_process.unique_pid': uniquePIDs,
+                            },
+                          },
+                        ],
+                      },
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          alerts: {
+            filter: { term: { 'event.kind': 'alert' } },
+            aggs: {
+              ids: { terms: { field: 'endgame.data.alert_details.acting_process.unique_pid' } },
+            },
+          },
+          events: {
+            filter: { term: { 'event.kind': 'event' } },
+            aggs: {
+              ids: { terms: { field: 'endgame.unique_pid' } },
+            },
+          },
+        },
+      },
+      index,
+    };
+  }
+
+  protected query(entityIDs: string[], index: string): JsonObject {
+    return {
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              { terms: { 'process.entity_id': entityIDs } },
+              {
+                bool: {
+                  should: [
+                    // event part of the filter
+                    {
+                      bool: {
+                        filter: [
+                          { term: { 'event.kind': 'event' } },
+                          {
+                            bool: {
+                              must_not: {
+                                term: { 'event.category': 'process' },
+                              },
+                            },
+                          },
+                        ],
+                      },
+                    },
+                    // alert part of the filter
+                    { term: { 'event.kind': 'alert' } },
+                  ],
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          alerts: {
+            filter: { term: { 'event.kind': 'alert' } },
+            aggs: {
+              ids: { terms: { field: 'process.entity_id' } },
```

Comment:
I'm not sure if this is working the way we are hoping. I think this defaults to 10 buckets, but I think the hope is to get a bucket for each unique entity_id. This can probably be addressed in a follow up PR.

Reply:
Perhaps we could just remove the stats part of this API. We aren't using it, right?

Reply:
Yeah I'm fine with that.